### PR TITLE
Add documentation for Tuple extraction in Scala DSL (#283)

### PR DESCRIPTION
**Pull request by richard-imaoka**
image

### CODE REVIEW

File: docs/src/main/paradox/scala/http/routing-dsl/directives/index.md [hunk 1]
```diff
@@ -231,4 +231,47 @@ val route =
...
+## Automatic Tuple extraction (flattening)
+
+Convenient Scala DSL syntax described in @ref[Basics](#basics), and @ref[Composing Directives](#composing-directives) 
+are made possible by Tuple extraction internally. Let's see how this works with examples.
+ 
+```scala
+val futureOfInt: Future[Int] = Future.successful(1)
+val route =
+  path("success") {
+    onSuccess(futureOfInt) { //: Dirctive[Tuple1[Int]]
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/routing-dsl/directives/index.md]: s/Dirctive/Directive/ typo in the three scala code blocks

Comment (richard-imaoka) [inline docs/src/main/paradox/scala/http/routing-dsl/directives/index.md]: Thanks for finding them out! I fixed them in the new commit.

File: docs/src/main/paradox/scala/http/routing-dsl/directives/index.md [hunk 2]
```diff
@@ -231,4 +231,47 @@ val route =
...
+```scala
+val futureOfInt: Future[Int] = Future.successful(1)
+val route =
+  path("success") {
+    onSuccess(futureOfInt) { //: Directive[Tuple1[Int]]
+      i => complete("Future was completed.")
+    }
+  }
+```
+Looking at the above code, onSuccess(futureOfInt) returns a Directive1[Int] = Directive[Tuple1[Int]].
```
Comment (ktoso) [inline docs/src/main/paradox/scala/http/routing-dsl/directives/index.md]: all code snippets should be in 

```
`onSuccess(futureOfInt)`
```

I think, please add?

**Comment by richard-imaoka** [comment]
Please see the updated screenshot of rendered HTML - does it look better?

image

**Comment by ktoso** [comment]
Great, thanks a lot @richard-imaoka !

**Comment by ktoso** [comment]
resolves #283

**Pull request merged**